subsys/bluetooth/host/sco_internal.h

```diff
@@ -0,0 +1,96 @@
+/** @file
+ *  @brief Internal APIs for Bluetooth SCO handling.
+ */
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @brief SCO Accept Info Structure */
+struct bt_sco_accept_info {
+	/** The ACL connection that is requesting authorization */
+	struct bt_conn *acl;
+
+	/** class code of peer device */
+	uint8_t   dev_class[3];
+
+	/** link type */
+	uint8_t   link_type;
+};
+
+/** @brief SCO Server structure. */
+struct bt_sco_server {
+#if defined(CONFIG_BT_SMP) || defined(__DOXYGEN__)
+	/** Required minimum security level.
+	 * Only available when @kconfig{CONFIG_BT_SMP} is enabled.
+	 */
+	bt_security_t		sec_level;
+#endif /* CONFIG_BT_SMP */
+	/** @brief Server accept callback
+	 *
+	 *  This callback is called whenever a new incoming connection requires
+	 *  authorization.
+	 *
+	 *  @param info The SCO accept information structure
+	 *  @param chan Pointer to receive the allocated channel
+	 *
+	 *  @return 0 in case of success or negative value in case of error.
+	 */
+	int (*accept)(const struct bt_sco_accept_info *info,
+			  struct bt_sco_chan **chan);
+};
+
+/** @brief Register SCO server.
+ *
+ *  Register SCO server, each new connection is authorized using the accept()
+ *  callback which in case of success shall allocate the channel structure
+ *  to be used by the new connection.
+ *
+ *  @param server Server structure.
+ *
+ *  @return 0 in case of success or negative value in case of error.
+ */
+int bt_sco_server_register(struct bt_sco_server *server);
```

Comment:
Don't address this in this PR, it's just something I think that should be done in the future.

To me that sounds like an API that should be public. I don't know enough about Classic, are users allowed to create SCO connections?
In the same vein, I see that in order to use HFP, etc.. you have to include private headers, ie the tests that test RFCOMM and other classic profiles need to include the corresponding private header for the profile API.

I believe moving those to a `include/bluetooth/classic/[profilename].h` would be a good thing for usability.
Unless they are not supposed to be configured from the app, tell me if I'm wrong @jhedberg .

Reply:
> I believe moving those to a `include/bluetooth/classic/[profilename].h` would be a good thing for usability. Unless they are not supposed to be configured from the app, tell me if I'm wrong @jhedberg.

Yes, I think we're overdue with having dedicated directories for BR/EDR implementation and header files. It'd also make the references in `MAINTAINERS.yml` simpler. So I'd propose to first move all existing BR/EDR-specific code to `subsys/bluetooth/host/classic/` and `include/bluetooth/classic/` before adding any new files or features.

Reply:
From my view, SCO and eSCO need to work together with HFP unit/HFP AG. So I think the source code of SCO/eSCO could be moved internally without exposing the interface. The SCO/eSCO connection will be opened or closed internally by the HFP unit/AG following the HFP sequence in a timely manner.

I will do it in this PR.

> Yes, I think we're overdue with having dedicated directories for BR/EDR implementation and header files. It'd also make the references in `MAINTAINERS.yml` simpler. So I'd propose to first move all existing BR/EDR-specific code to `subsys/bluetooth/host/classic/` and `include/bluetooth/classic/` before adding any new files or features.
Yes. What can I do to help make this change happen?

Reply:
You can just make another PR moving the files (and updating the paths in include, cmake and Maintainers)